api/src/display.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+//! Character encodings for public addresses and request types for easy
+//! cut-and-paste.
+
+use crate::printable;
+use crc::crc32;
+use failure::Fail;
+use protobuf::{parse_from_bytes, Message};
+
+/// Decoding / encoding errors
+#[derive(Clone, Debug, Eq, PartialEq, Fail)]
+pub enum Error {
+    /// Protocol buffer could not be serialized
+    #[fail(display = "Could not serialize protocol buffer: {:?}", _0)]
+    SerializationFailure(String),
+
+    /// The b58 string cannot be converted into bytes
+    #[fail(display = "Could not decode b58: {:?}", _0)]
+    B58Error(String),
+
+    /// Protocol buffer could not be deserialized
+    #[fail(display = "Could not deserialize protocol buffer: {:?}", _0)]
+    DeserializationFailure(String),
+
+    /// Checksum does not match
+    #[fail(display = "Checksum does not match payload")]
+    ChecksumFailure(),
+}
+
+/// A little-endian IEEE CRC32 checksum is prepended to payloads.
```

Comment:
can we add a comment here explaining that we believe CRC32 to be an improvement over the [bitcoin standard](https://en.bitcoin.it/wiki/Base58Check_encoding)?